File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
const ALPHABET = "abcdefghijklmnopqrstuvwxyz";

function solution(s, skip, index) {
    let answer = '';
    
    const validAlphabet = ALPHABET.split("").filter(char => !skip.includes(char));
    
    for (let char of s) {
        const currentIdx = validAlphabet.indexOf(char);
        const newIdx = (currentIdx + index) % validAlphabet.length;
        answer += validAlphabet[newIdx];
    }
    
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
import string

def solution(s, skip, index):
    answer = ''
    
    alphabet = string.ascii_lowercase
    filtered_alpha = [char for char in alphabet if char not in skip]
    
    for char in s:
        current_idx = filtered_alpha.index(char)
        new_idx = (current_idx + index) % len(filtered_alpha)
        answer += filtered_alpha[new_idx]
    
    return answer